### PR TITLE
[TE] Creating a thirdeye-dashboard module to host the dashboard server

### PR DESCRIPTION
**Pull request by suvodeep-pyne**
## Module refactor changes
- Moved `org.apache.pinot.thirdeye.dashboard` from `thirdeye-pinot` to `thirdeye-dashboard`
- Moved all test code from `thirdeye-pinot` to `thirdeye-dashboard`
- Moved config files from `thirdeye-pinot` to `thirdeye-dashboard`
- Module `thirdeye-pinot` no longer builds a shaded jar. 
- The `dashboard` module builds a shaded jar.
- The launcher scripts now point to the dashboard module

This is a **BREAKING** change.

The change is as per the new Module architecture doc shared on the thirdeye slack. 

At a later stage, the more modular unit tests will be refactored to colocate them with their respective classes. This is because, currently there are a lot of tests which use a lot to classes to run thereby behaving more like integration tests as opposed to unit tests. It might be worth considering having a `thirdeye-integration-tests` module for these cases.

### CODE REVIEW

**Review by apucher — COMMENTED** [review]
looks worse than it is. 👍 I'd check the encoding of some of the csv files, looks ike a lot of whitespace changes.

**Comment by suvodeep-pyne** [comment]
I think IntelliJ did some magic when moving those files. It is fixing the line separators as per *nix standards.

File: thirdeye/thirdeye-pinot/pom.xml [hunk 1]
```diff
@@ -361,52 +361,6 @@
 
   <build>
     <plugins>
-      <plugin>
-        <artifactId>maven-shade-plugin</artifactId>
```
Comment (vincentchenjl) [inline thirdeye/thirdeye-pinot/pom.xml:365]: Per our discussion, we are going to publish shaded JAR and source JAR together. We should only remove this plugin once we have both JAR build.

Comment (suvodeep-pyne) [reply thirdeye/thirdeye-pinot/pom.xml:365]: The shading is moved to the `thirdeye-dashboard`.

**Review by vincentchenjl — APPROVED** [review]
LGTM! Thanks for the contribution!

**Review by jihaozh — APPROVED** [review]
Awesome, LGTM

**Comment by suvodeep-pyne** [comment]
Thanks a lot @vincentchenjl @jihaozh !

**Pull request merged**